fastify.js

```diff
@@ -792,6 +792,7 @@ function fastify (options) {
     let isAsync
     return function preRouting (req, res) {
       // only call isAsyncConstraint once
+      req.originalUrl = req.originalUrl || req.url
```

Comment:
Can you move this line within the `if (rewriteUrl)` block and then move the `||` in https://github.com/fastify/fastify/pull/4758/files#diff-cf6996cafd1df0bc168a5c5f3a7470f11fc3aea9f908091d5dd6be2e7b30a2c3R471.

This will otherwise generate a new hidden class for the `req` object unnecessarily.